code/controllers/Processes/timer.dm

```diff
@@ -98,3 +98,11 @@ DECLARE_GLOBAL_CONTROLLER(timer, timer_master)
 			qdel(event)
 			return 1
 	return 0
+
+/proc/getRemainingTime(timer)
+	if(!timer)
+		return FALSE
+	for(var/datum/timedevent/event in timer_master.processing_timers)
```

Comment:
This is my biggest concern here.

As we've become more reliant on timers, the cost of the overall system has gone up; innately, what the controller does isn't all that expensive, but since it has to loop through every timer in existence, it adds up--likewise, checking a reference timer requires you to do the same thing.

If we had TG's updated timer system, I'd say that something like this, only an associative list, would probably be ok (though I'd still question it use and want to know specific).